includes/reports/class-sensei-no-users-table-relationship.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * File containing the Sensei_No_Users_Table_Relationship class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class that handles a special case of environments that have the users table in a separate
+ * database from the other tables.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_No_Users_Table_Relationship {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_No_Users_Table_Relationship constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
```

Comment:
Technically this method should have `@access private`, right?

Reply:
I don't think so, since we want to call it from another class. I think we just use the `@access private` for cases that is really private, but we need to expose for some special reason, like the hooks.

But let me know if I'm missing anything here, so we can change it. ;)

Reply:
Fair enough. I just saw that we don't add `@access private` to init methods in general. Thanks for the reply!